azkaban-common/src/main/java/azkaban/server/AzkabanServer.java

```diff
@@ -75,12 +77,27 @@ public static Props loadProps(String[] args, OptionParser parser) {
       azkabanSettings = loadConfigurationFromAzkabanHome();
     }
 
+    if (azkabanSettings != null) {
```

Comment:
What's the advantage?

Shall we throw an exception if the setting is null here? 
If so, I would suggest doing it in another pull request to keep refactoring and functional change separate.